src/main/java/ch/njol/skript/effects/EffPlaySound.java

```diff
@@ -51,182 +50,127 @@
 		"play sound \"entity.experience_orb.pickup\" with volume 0.5 to the player",
 		"play sound \"custom.music.1\" in jukebox category at {speakerBlock}"})
 @Since("2.2-dev28, 2.4 (sound categories)")
-@RequiredPlugins("Minecraft 1.11+ (sound categories)")
 public class EffPlaySound extends Effect {
 
-	private static final boolean SOUND_CATEGORIES_EXIST = Skript.classExists("org.bukkit.SoundCategory");
-	private static final Pattern SOUND_VALID_PATTERN = Pattern.compile("[a-z0-9\\/:._-]+"); // Minecraft only accepts these characters
+	private static final Pattern KEY_PATTERN = Pattern.compile("([a-z0-9._-]+:)?[a-z0-9/._-]+");
 	
 	static {
-		if (SOUND_CATEGORIES_EXIST) {
-			Skript.registerEffect(EffPlaySound.class,
-					"play sound[s] %strings% [(in|from) %-soundcategory%] " +
-							"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] at %locations% [(to|for) %-players%]",
-					"play sound[s] %strings% [(in|from) %-soundcategory%] " +
-							"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] [(to|for) %players%] [(at|from) %-locations%]");
-		} else {
-			Skript.registerEffect(EffPlaySound.class,
-					"play sound[s] %strings% [(at|with) volume %-number%] " +
-							"[(and|at|with) pitch %-number%] at %locations% [(to|for) %-players%]",
-					"play sound[s] %strings% [(at|with) volume %-number%] " +
-							"[(and|at|with) pitch %-number%] [(to|for) %players%] [(at|from) %-locations%]");
-		}
+		Skript.registerEffect(EffPlaySound.class,
+				"play sound[s] %strings% [(in|from) %-soundcategory%] " +
+						"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] at %locations% [(to|for) %-players%]",
+				"play sound[s] %strings% [(in|from) %-soundcategory%] " +
+						"[(at|with) volume %-number%] [(and|at|with) pitch %-number%] [(to|for) %players%] [(at|from) %-locations%]");
 	}
 
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<String> sounds;
-	@Nullable
-	private Expression<SoundCategory> category;
-	@Nullable
-	private Expression<Number> volume, pitch;
-	@Nullable
-	private Expression<Location> locations;
-	@Nullable
-	private Expression<Player> players;
+	private @Nullable Expression<SoundCategory> category;
+	private @Nullable Expression<Number> volume;
+	private @Nullable Expression<Number> pitch;
+	private @Nullable Expression<Location> locations;
+	private @Nullable Expression<Player> players;
```

Comment:
We keep the nullable annotation above the field its representing, you can place new line breaks between each if you want to make it look cleaner.